core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -2113,6 +2113,29 @@ private void checkExponentialCnf(int n) {
     checkSimplify(rexNode, "false");
   }
 
+  @Test void testSimplifyAndIsDistinctFrom() {
```

Comment:
Add jira link here maybe better.

Reply:
Fixed in https://github.com/apache/calcite/pull/4525/commits/0f847ffcd4650a68390e584f54ec23e2307850f1